### PR TITLE
[sharktank] Materialize indices in PrefillTask to ease branching logic in downstream code

### PR DESCRIPTION
**Pull request by sogartar**
Handling the the None special case is error prone and adds branching complexity. We don't care that much about performance when doing evaluation.

Allow compute_cross_entropy to accept sequences of logits and indicies instead of single ndarray. This makes the function symmetric with respect to the requests arg. Which is a sequence.
This will help with chunked prefill and other methods of constructing the results where they are stitched from multiple requests. We don't need to burden the caller to make a single ndarray when we don't need it. But a single ndarray is still supported.

### CODE REVIEW

File: sharktank/sharktank/utils/llm_utils.py [hunk 1]
```diff
@@ -702,19 +702,19 @@ def __init__(self, batch: LlmRunner, logits_normalization: str):
         self._logits_normalization = logits_normalization
 
     def compute_cross_entropy(
-        self, logits, indices, requests, min_context=0
+        self,
+        logits: list[numpy.ndarray],
+        indices: list[numpy.ndarray],
+        requests: list[list[int]],
+        min_context=0,
```
Comment (archana-ramalingam) [inline sharktank/sharktank/utils/llm_utils.py]: Add type hint to last arg too.

File: sharktank/tests/utils/llm_prefill_tasks_test.py [hunk 2]
```diff
@@ -201,7 +201,7 @@ def _invocation_fn(*args):
             logits,
             numpy.array([i for i in range(self._batch_size)], dtype=self._int_dtype),
         )
-        assert indices is None
+        assert numpy.array_equal(indices, [0, 1, 2, 3])
```
Comment (archana-ramalingam) [inline sharktank/tests/utils/llm_prefill_tasks_test.py]: Same comment as above.

File: sharktank/tests/utils/llm_prefill_tasks_test.py [hunk 3]
```diff
@@ -335,4 +335,4 @@ def _invocation_fn(*args):
...
```
Comment (archana-ramalingam) [inline sharktank/tests/utils/llm_prefill_tasks_test.py]: Same comment as above.

File: sharktank/tests/utils/llm_prefill_tasks_test.py [hunk 4]
```diff
@@ -123,7 +123,7 @@ def _invocation_fn(*args):
...
```
[the remark this answers is not in the document]
Comment (sogartar) [reply sharktank/tests/utils/llm_prefill_tasks_test.py]: Done.

File: sharktank/tests/utils/llm_prefill_tasks_test.py (hunk 2 above)
Comment (sogartar) [reply sharktank/tests/utils/llm_prefill_tasks_test.py]: Done.

File: sharktank/tests/utils/llm_prefill_tasks_test.py (hunk 3 above)
Comment (sogartar) [reply sharktank/tests/utils/llm_prefill_tasks_test.py]: Done.
